sql/core/src/test/scala/org/apache/spark/sql/CTEInlineSuite.scala

```diff
@@ -678,6 +678,17 @@ abstract class CTEInlineSuiteBase
       }.isDefined, "CTE columns should not be pruned.")
     }
   }
+
+  test("SPARK-45752: Unreferenced CTE should all be checked by CheckAnalysis0") {
+    val e = intercept[AnalysisException](sql(
+      s"""
+        |with
+        |a as (select * from non_exist),
+        |b as (select * from a)
+        |select 2
+        |""".stripMargin))
+    assert(e.getMessage.contains("[TABLE_OR_VIEW_NOT_FOUND]"))
```

Comment:
we should use the `checkError` testing function.

Reply:
+1